### PR TITLE
Added `Interfaces.md` in the kotlin folder

### LINKED ISSUE #1949: [Team Entry] Interfaces: Adding Interfaces.md in the kotlin folder
### Reason/inspiration (optional)

The entry will have:
- Have a section on description of interfaces.
- Have a syntax pseudo block for interfaces.
- Have an example code block for interfaces.

### Entry Type (select all that apply)

New Topic Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [X] Issue is assigned

### PR DESCRIPTION
**Pull request by dakshdeepHERE**
### Type of Change

- Adding a new entry
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

Fixes #1949

### CODE REVIEW

File: content/kotlin/concepts/Interfaces/Interfaces.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+---
+Title: 'Interface'
+Description: 'Interface are like abstract class describing methods and variables that should exist in any class that implements the interface.'
```
Comment (SSwiniarski) [inline content/kotlin/concepts/Interfaces/Interfaces.md:3]:
```suggestion
Description: 'Interfaces are an abstract class describing methods and variables that should exist in any class that implements the interface.'
```

File: content/kotlin/concepts/Interfaces/Interfaces.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+Tags:
+  - 'Classes'
+  - 'Inheritance'
+  - 'OOP'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+An **interface** is a type that defines a set of methods and properties that a class can implement. It provides a way to define a contract between the implementing class and the calling code. An interface can be thought of as a blueprint for a class, where the class must implement all the methods and properties defined in the **interface**.
```
Comment (SSwiniarski) [inline content/kotlin/concepts/Interfaces/Interfaces.md:15]:
```suggestion
An **interface** is a type that defines a set of methods and properties that a class can implement. It provides a way to define a contract between the implementing class and the calling code. An interface can be thought of as a blueprint for a class, where the class must implement all the methods and properties defined in the interface.
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@dakshdeepHERE I did a first review and left a couple of comments.

**Comment by dakshdeepHERE** [comment]
done @SSwiniarski

**Review by SSwiniarski — APPROVED** [review]
@dakshdeepHERE Looks good for a second review.

**Comment by yangc95** [comment]
🙌 @dakshdeepHERE live link: https://www.codecademy.com/resources/docs/kotlin/interfaces

cc: @KTom101 @SSwiniarski

**Comment by yangc95** [comment]
@dakshdeepHERE Actually, it looks like the folder and file names are capitalized instead of lower-cased. Can you open a new PR and adjust this? Thanks!

**Comment by dakshdeepHERE** [comment]
@yangc95 I did small in start only if you check the commits but I checked the java folder there it was starting with "I" so K recommitted it.

**Comment by yangc95** [comment]
@dakshdeepHERE I don't see any commits on your feature branch. Would you be able to create a PR and we can go from there?

**Comment by dakshdeepHERE** [comment]
@yangc95 yes I renamed the folder but see when I do git status there is no changes shown I dont know why
image
image
Why is that?

**Comment by dakshdeepHERE** [comment]
see even the file name changes arent showing as changes 
image
image

**Comment by yangc95** [comment]
@dakshdeepHERE Change the file metadata from the title "Interface" to "Interfaces" and push up to create the PR. I can help from there with the renaming

**Comment by dakshdeepHERE** [comment]
@yangc95 Oh okay! I got it will do it right now.

**Comment by yangc95** [comment]
Thanks @dakshdeepHERE ! [#2076](https://github.com/Codecademy/docs/pull/2076) is merged now and the entry should be live at the correct link: https://www.codecademy.com/resources/docs/kotlin/interfaces

**Pull request merged**